openstf/feature_engineering/apply_features.py

```diff
@@ -149,185 +55,50 @@ def apply_features(data, minute_list=None, h_ahead=24):
                             np.random.uniform(0.7,1.7, 200)))
 
     """
-    if minute_list is None:
-        minute_list = []
 
-    lag_functions = generate_lag_functions(data, minute_list, h_ahead)
+    # Get lag feature functions
+    lag_feature_functions = generate_lag_feature_functions(feature_set_list, horizon)
 
-    # TODO it seems not all feature use the same convention
-    # better to choose one and stick to it
-    timedriven_functions = {
+    # Get timedrivenfeature functions
+    timedriven_feature_functions = {
         "IsWeekendDay": lambda x: (x.index.weekday // 5) == 1,
         "IsWeekDay": lambda x: x.index.weekday < 5,
-        "IsMonday": lambda x: x.index.weekday == 0,
-        "IsTuesday": lambda x: x.index.weekday == 1,
-        "IsWednesday": lambda x: x.index.weekday == 2,
-        "IsThursday": lambda x: x.index.weekday == 3,
-        "IsFriday": lambda x: x.index.weekday == 4,
-        "IsSaturday": lambda x: x.index.weekday == 5,
         "IsSunday": lambda x: x.index.weekday == 6,
         "Month": lambda x: x.index.month,
         "Quarter": lambda x: x.index.quarter,
     }
 
-    # Add check for specific hour
-    for thour in np.linspace(0, 23, 24):
-
-        def func(x, checkhour=thour):
-            return x.index.hour == checkhour
+    # Get holiday feature functions
+    holiday_feature_functions = create_holiday_feature_functions()
 
-        char = "0" if thour < 10 else ""
-        new = {"Is" + char + str(int(thour)) + "Hour": func}
-        timedriven_functions.update(new)
-
-    df = data.copy()
-    # holiday_function are imported at the beginning of the file and includes all dutch
-    # school/work holidays
-    holiday_functions = create_holiday_functions()
+    # Only select features that occur in feature_set_list,
+    # if feature_set_list is none nothing is removed
+    if feature_set_list is not None:
```

Comment:
Perhaps its simpler to remove this code and just skip the keys not in `feature_set_list` at line 95.